### PR TITLE
feat: installing csi-proxy for windows at node deployment time

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Redo of #2854
Add support for installing csi-proxy for windows at node-deloyment time.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by marosset** [comment]
/cc @kkmsft @andyzhangx @jackfrancis @mboersma 
Re-doing this PR since it got merged pre-maturely due to bad Tide config.

I also updated the powershell to set kubelet service to depend on csi-proxy-server service if enabled (per KK's feedback on other PR)

File: pkg/api/vlabs/types.go [hunk 1]
```diff
@@ -162,6 +162,8 @@ type CustomNodesDNS struct {
 type WindowsProfile struct {
 	AdminUsername          string            `json:"adminUsername,omitempty"`
 	AdminPassword          string            `json:"adminPassword,omitempty"`
+	CSIProxyURL            string            `json:"csiProxyURL"`
```
Comment (jackfrancis) [inline pkg/api/vlabs/types.go]: I think we should `omitempty` here. I've also noticed other properties are probably missing this, and filed a backlog issue here:

https://github.com/Azure/aks-engine/issues/2933

Comment (marosset) [reply pkg/api/vlabs/types.go]: @jackfrancis should I add omitempty only for vlabs/types.go or pkg/api/types.go as well?

Comment (jackfrancis) [reply pkg/api/vlabs/types.go]: to be consistent and cautious I always duplicate it in both versions, tho AFAIK the vlabs is the only one that matters in terms of user api model impact

Comment (marosset) [reply pkg/api/vlabs/types.go]: k - will update

Comment (marosset) [reply pkg/api/vlabs/types.go]: updated

**Pull request merged**